torch/utils/data/_utils/pin_memory.py

```diff
@@ -77,12 +77,11 @@ def pin_memory(data, device=None):
             # The mapping type may not support `copy()` / `update(mapping)`
             # or `__init__(iterable)`.
             return {k: pin_memory(sample, device) for k, sample in data.items()}
-    elif isinstance(data, tuple) and hasattr(data, "_fields"):  # namedtuple
-        return type(data)(*(pin_memory(sample, device) for sample in data))
     elif isinstance(data, tuple):
-        return [
-            pin_memory(sample, device) for sample in data
-        ]  # Backwards compatibility.
+        if hasattr(data, "_fields"):  # namedtuple
+            return type(data)(*(pin_memory(sample, device) for sample in data))
+        else:
+            return type(data)(pin_memory(sample, device) for sample in data)
```

Comment:
Wow, I totes didn't realise that tuple and namedtuple do NOT have compatible constructors.  That's awkward.